static/app/routes.tsx

```diff
@@ -148,6 +411,43 @@ function buildRoutes() {
     </Route>
   ) : null;
 
+  /*
+    This component mounts existing routes with a lower precedence than the new
+    custom domain routes. It will also redirect to the new routes if the
+    custom domains feature flag is turned on for this organization. Replacements look like:
+
+    sentry.io/organizations/{slug}/issues/ -> {slug}.sentry.io/issues/
+    sentry.io/settings/{slug}/ -> {slug}.sentry.io/settings/
+    sentry.io/onboarding/{slug}/{step} -> {slug}.sentry.io/onboarding/{step}
+    sentry.io/settings/{slug}/projects/{projectslug}/ -> {slug}.sentry.io/settings/projects/{project.slug}/
+  */
+  const orgScopedCustomDomainRoute = originalRoute => {
+    const newPath = originalRoute.props.path
+      .replace(/organizations\/:orgId\/?/, '')
+      .replace(/:orgId\/?/, '');
+    const customDomainRoute = (
+      <Route path={newPath} component={originalRoute.component} />
```

Comment:
We will need to copy `props` from the existing routes too. Several issue related routes have `props` defined.

We might also need a branch to handle `IndexRoute` instances and existing `Redirect` routes.

Reply:
I revamped this. I ended up having to put together a full tree traversal. It's not quite working yet, but I think it's pretty close